src/fsharp/ConstraintSolver.fs

```diff
@@ -1914,67 +1914,73 @@ and CanMemberSigsMatchUpToCheck
     let unnamedCalledOutArgs = calledMeth.UnnamedCalledOutArgs
 
     // First equate the method instantiation (if any) with the method type parameters 
-    if minst.Length <> uminst.Length then ErrorD(Error(FSComp.SR.csTypeInstantiationLengthMismatch(), m)) else
-    
-    Iterate2D unifyTypes minst uminst ++ (fun () -> 
-
-    if not (permitOptArgs || isNil unnamedCalledOptArgs) then ErrorD(Error(FSComp.SR.csOptionalArgumentNotPermittedHere(), m)) else
-    
-
-    let calledObjArgTys = calledMeth.CalledObjArgTys(m)
+    if minst.Length <> uminst.Length then 
+        ErrorD(Error(FSComp.SR.csTypeInstantiationLengthMismatch(), m))
+    else
+        trackErrors {
+            do! Iterate2D unifyTypes minst uminst
+            do!
+                if not (permitOptArgs || isNil unnamedCalledOptArgs) then 
```

Comment:
I believe you don't need the `do!`, you can go straight into the `if` and then `return!` on each branch
```fsharp
        trackErrors {
            do! Iterate2D unifyTypes minst uminst
            if not (permitOptArgs || isNil unnamedCalledOptArgs) then 
                return! ErrorD(Error(FSComp.SR.csOptionalArgumentNotPermittedHere(), m)) 
            else
             ....
```